pkg/p2p/peer/router.go

```diff
@@ -54,47 +55,61 @@ func (m *messageRouter) CanRoute(topic topics.Topic) bool {
 	return false
 }
 
-func (m *messageRouter) route(topic topics.Topic, b *bytes.Buffer) {
+func (m *messageRouter) route(b bytes.Buffer, msg message.Message) error {
 	var err error
-	switch topic {
+	category := msg.Category()
+	switch category {
 	case topics.GetBlocks:
-		err = m.blockHashBroker.AdvertiseMissingBlocks(b)
+		topics.Extract(&b)
+		err = m.blockHashBroker.AdvertiseMissingBlocks(&b)
 	case topics.GetData:
-		err = m.dataBroker.SendItems(b)
+		topics.Extract(&b)
+		err = m.dataBroker.SendItems(&b)
 	case topics.MemPool:
+		topics.Extract(&b)
 		err = m.dataBroker.SendTxsItems()
 	case topics.Inv:
-		err = m.dataRequestor.RequestMissingItems(b)
+		topics.Extract(&b)
+		err = m.dataRequestor.RequestMissingItems(&b)
 	case topics.Block:
-		err = m.synchronizer.Synchronize(b, m.peerInfo)
+		topics.Extract(&b)
+		err = m.synchronizer.Synchronize(&b, m.peerInfo)
 	case topics.Ping:
+		topics.Extract(&b)
```

Comment:
Could we instead, extract the topic before going into the `switch` statement? To avoid some duplicate code